xls/examples/lfsr.x

```diff
@@ -0,0 +1,438 @@
+    let _ = assert_eq(lfsr7(u7:73), u7:19);
+    let _ = assert_eq(lfsr7(u7:19), u7:38);
+    let _ = assert_eq(lfsr7(u7:38), u7:77);
+    let _ = assert_eq(lfsr7(u7:77), u7:27);
+    let _ = assert_eq(lfsr7(u7:27), u7:54);
+    let _ = assert_eq(lfsr7(u7:54), u7:109);
+    let _ = assert_eq(lfsr7(u7:109), u7:90);
+    let _ = assert_eq(lfsr7(u7:90), u7:53);
+    let _ = assert_eq(lfsr7(u7:53), u7:107);
+    let _ = assert_eq(lfsr7(u7:107), u7:86);
+    let _ = assert_eq(lfsr7(u7:86), u7:45);
+    let _ = assert_eq(lfsr7(u7:45), u7:91);
+    let _ = assert_eq(lfsr7(u7:91), u7:55);
+    let _ = assert_eq(lfsr7(u7:55), u7:111);
+    let _ = assert_eq(lfsr7(u7:111), u7:94);
+    let _ = assert_eq(lfsr7(u7:94), u7:61);
+    let _ = assert_eq(lfsr7(u7:61), u7:123);
+    let _ = assert_eq(lfsr7(u7:123), u7:118);
+    let _ = assert_eq(lfsr7(u7:118), u7:108);
+    let _ = assert_eq(lfsr7(u7:108), u7:88);
+    let _ = assert_eq(lfsr7(u7:88), u7:49);
+    let _ = assert_eq(lfsr7(u7:49), u7:99);
+    let _ = assert_eq(lfsr7(u7:99), u7:70);
+    let _ = assert_eq(lfsr7(u7:70), u7:13);
+    let _ = assert_eq(lfsr7(u7:13), u7:26);
+    let _ = assert_eq(lfsr7(u7:26), u7:52);
+    let _ = assert_eq(lfsr7(u7:52), u7:105);
+    let _ = assert_eq(lfsr7(u7:105), u7:82);
+    let _ = assert_eq(lfsr7(u7:82), u7:37);
+    let _ = assert_eq(lfsr7(u7:37), u7:75);
+    let _ = assert_eq(lfsr7(u7:75), u7:23);
+    let _ = assert_eq(lfsr7(u7:23), u7:46);
+    let _ = assert_eq(lfsr7(u7:46), u7:93);
+    let _ = assert_eq(lfsr7(u7:93), u7:59);
+    let _ = assert_eq(lfsr7(u7:59), u7:119);
+    let _ = assert_eq(lfsr7(u7:119), u7:110);
+    let _ = assert_eq(lfsr7(u7:110), u7:92);
+    let _ = assert_eq(lfsr7(u7:92), u7:57);
+    let _ = assert_eq(lfsr7(u7:57), u7:115);
+    let _ = assert_eq(lfsr7(u7:115), u7:102);
+    let _ = assert_eq(lfsr7(u7:102), u7:76);
+    let _ = assert_eq(lfsr7(u7:76), u7:25);
+    let _ = assert_eq(lfsr7(u7:25), u7:50);
+    let _ = assert_eq(lfsr7(u7:50), u7:101);
+    let _ = assert_eq(lfsr7(u7:101), u7:74);
+    let _ = assert_eq(lfsr7(u7:74), u7:21);
+    let _ = assert_eq(lfsr7(u7:21), u7:42);
+    let _ = assert_eq(lfsr7(u7:42), u7:85);
+    let _ = assert_eq(lfsr7(u7:85), u7:43);
+    let _ = assert_eq(lfsr7(u7:43), u7:87);
+    let _ = assert_eq(lfsr7(u7:87), u7:47);
+    let _ = assert_eq(lfsr7(u7:47), u7:95);
+    let _ = assert_eq(lfsr7(u7:95), u7:63);
+    let _ = assert_eq(lfsr7(u7:63), u7:127);
+    let _ = assert_eq(lfsr7(u7:127), u7:126);
+    let _ = assert_eq(lfsr7(u7:126), u7:124);
+    let _ = assert_eq(lfsr7(u7:124), u7:120);
+    let _ = assert_eq(lfsr7(u7:120), u7:112);
+    let _ = assert_eq(lfsr7(u7:112), u7:96);
+    let _ = assert_eq(lfsr7(u7:96), u7:64);
+    let _ = assert_eq(lfsr7(u7:64), u7:1);
+	()
+}
+
+#[test]
+fn lfsr8_test() {
+	let _ = assert_eq(lfsr8(u8:0), u8:0);
+	let _ = assert_eq(lfsr8(u8:1), u8:2);
+	let _ = assert_eq(lfsr8(u8:2), u8:4);
+	let _ = assert_eq(lfsr8(u8:4), u8:8);
```

Comment:
if we want to keep only dslx test, maybe we could make thiose more succint to ease maintainance (test the first N values and then do noop iteration for `PERIOD - N` and test the next `N` value to verifie it does cycle?

wdyt?

Reply:
Yes, I changed the tests to only include the sanity test (with 0), a few random values and the cycle.